### PR TITLE
[jvm-packages] Implement new `save_raw` in jvm-packages.

### PR DESCRIPTION
**Pull request by trivialfis**
From https://github.com/dmlc/xgboost/pull/7545 .

### CODE REVIEW

File: jvm-packages/xgboost4j/src/main/scala/ml/dmlc/xgboost4j/scala/Booster.scala [hunk 1]
```diff
@@ -301,6 +301,19 @@ class Booster private[xgboost4j](private[xgboost4j] var booster: JBooster)
 
   def getVersion: Int = booster.getVersion
 
+  /**
+    * Save model into a raw byte array.  Available options are "json", "ubj" and "deprecated".
+    */
+  @throws(classOf[XGBoostError])
+  def toByteArray(format: String): Array[Byte] = {
```
Comment (wbo4958) [inline jvm-packages/xgboost4j/src/main/scala/ml/dmlc/xgboost4j/scala/Booster.scala:308]: Could we really expose this API to users?

File: jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/Booster.java [hunk 2]
```diff
@@ -636,13 +636,26 @@ public void setVersion(int version) {
...
+  }
+  /**
+   * Save model into raw byte array.
+   *
+   * @param format The output format.  Available options are "json", "ubj" and "deprecated".
+   *
+   * @return the saved byte array
+   * @throws XGBoostError native error
+   */
+  public byte[] toByteArray(String format) throws XGBoostError {
```
Comment (wbo4958) [inline jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/Booster.java:656]: Could we really expose this API to users?

File: jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/Booster.java [hunk 3]
```diff
@@ -636,13 +636,26 @@ public void setVersion(int version) {
...
+   * Save model into raw byte array. Currently it's using the deprecated format as
+   * default, which will be changed into `ubj` in future releases.
    *
-   * @return the saved byte array.
+   * @return the saved byte array
    * @throws XGBoostError native error
    */
   public byte[] toByteArray() throws XGBoostError {
+    return this.toByteArray("deprecated");
+  }
```
Comment (wbo4958) [inline jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/Booster.java:647]: Please add the empty line

**Review by wbo4958 — APPROVED** [review]
Overall, LGTM.

File: jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/Booster.java (hunk 3 above)
Comment (trivialfis) [reply jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/Booster.java:647]: ?

**Pull request merged**